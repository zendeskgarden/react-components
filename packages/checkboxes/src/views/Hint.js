/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import styled from 'styled-components';
import classNames from 'classnames';
import { retrieveTheme, isRtl } from '@zendeskgarden/react-theming';
import CheckboxStyles from '@zendeskgarden/css-forms/dist/checkbox.css';

const COMPONENT_ID = 'checkboxes.hint';

/**
 * Accepts all `<div>` props
 */
const Hint = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  className: props =>
    classNames(CheckboxStyles['c-chk__hint'], {
      // RTL
      [CheckboxStyles['is-rtl']]: isRtl(props)
    })
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default Hint;
