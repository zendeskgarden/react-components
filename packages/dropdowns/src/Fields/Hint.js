/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import { StyledHint } from '../styled';

/**
 * Accepts all `div` props. Must be nested with a `<Field>` component.
 */
const Hint = StyledHint;

Hint.displayName = 'Hint';

Hint.propTypes = {
  small: PropTypes.bool
};

export default Hint;
