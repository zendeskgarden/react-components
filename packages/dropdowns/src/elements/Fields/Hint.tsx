/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import { Hint as FormHint } from '@zendeskgarden/react-forms';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Hint = React.forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
  (props, ref) => <FormHint ref={ref} {...props} />
);

Hint.displayName = 'Hint';
