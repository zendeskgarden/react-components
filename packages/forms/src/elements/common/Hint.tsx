/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import useFieldContext from '../../utils/useFieldContext';
import useInputContext from '../../utils/useInputContext';
import { StyledHint, StyledCheckHint, StyledRadioHint, StyledToggleHint } from '../../styled';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Hint = React.forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
  (props, ref) => {
    const fieldContext = useFieldContext();
    const type = useInputContext();

    let HintComponent;

    if (type === 'checkbox') {
      HintComponent = StyledCheckHint;
    } else if (type === 'radio') {
      HintComponent = StyledRadioHint;
    } else if (type === 'toggle') {
      HintComponent = StyledToggleHint;
    } else {
      HintComponent = StyledHint;
    }

    let combinedProps = props;

    if (fieldContext) {
      combinedProps = fieldContext.getHintProps(combinedProps);
    }

    return <HintComponent ref={ref} {...(combinedProps as any)} />;
  }
);

Hint.displayName = 'Hint';
