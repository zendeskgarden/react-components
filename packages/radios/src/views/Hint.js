import styled from 'styled-components';
import classNames from 'classnames';
import { retrieveTheme, isRtl } from '@zendesk/garden-react-theming';
import CheckboxStyles from '@zendesk/garden-css-forms/dist/checkbox.css';

import { version } from '../../package.json';
const COMPONENT_ID = 'radios.hint';

/**
 * Accepts all `<div>` props
 */
const Hint = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version,
  className: props =>
    classNames(CheckboxStyles['c-chk__hint'], CheckboxStyles['c-chk__hint--radio'], {
      // RTL
      [CheckboxStyles['is-rtl']]: isRtl(props)
    })
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

/** @component */
export default Hint;
