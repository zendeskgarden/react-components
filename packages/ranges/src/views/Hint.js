/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';
import classNames from 'classnames';
import { retrieveTheme, isRtl } from '@zendeskgarden/react-theming';
import RangeStyles from '@zendeskgarden/css-forms/dist/range.css';

const COMPONENT_ID = 'ranges.hint';

/**
 * Accepts all `<div>` props
 */
const Hint = styled.div.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  className: props =>
    classNames(RangeStyles['c-range__hint'], {
      [RangeStyles['c-range__hint--sm']]: props.small,

      // RTL
      [RangeStyles['is-rtl']]: isRtl(props)
    })
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

Hint.propTypes = {
  small: PropTypes.bool
};

/** @component */
export default Hint;
