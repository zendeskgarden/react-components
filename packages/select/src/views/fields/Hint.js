import PropTypes from 'prop-types';
import { retrieveTheme } from '@zendesk/garden-react-theming';
import { Hint as TextfieldHint } from '@zendesk/garden-react-textfields';

import { version } from '../../../package.json';
const COMPONENT_ID = 'select.hint';

/**
 * Accepts all `<div>` props
 */
const Hint = TextfieldHint.extend.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': version
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

Hint.propTypes = {
  small: PropTypes.bool
};

/** @component */
export default Hint;
