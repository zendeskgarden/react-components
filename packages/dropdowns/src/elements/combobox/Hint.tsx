/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes, forwardRef, useEffect } from 'react';
import useFieldContext from '../../context/useFieldContext';
import { StyledHint } from '../../views';

/**
 * @deprecated use `Field.Hint` instead
 *
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Hint = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>((props, ref) => {
  const { hintProps, setHasHint } = useFieldContext();

  useEffect(() => {
    setHasHint(true);

    return () => setHasHint(false);
  }, [setHasHint]);

  return <StyledHint {...hintProps} {...props} ref={ref} />;
});

Hint.displayName = 'Hint';
