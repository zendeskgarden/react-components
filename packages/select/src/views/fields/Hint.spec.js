import React from 'react';
import { shallow } from 'enzyme';
import Hint from './Hint';

describe('Hint', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<Hint />);

    expect(wrapper).toMatchSnapshot();
  });
});
