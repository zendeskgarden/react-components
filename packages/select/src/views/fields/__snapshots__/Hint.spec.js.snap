// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint renders default styling 1`] = `
<div
  className="c-txt__hint "
  data-garden-id="select.hint"
  data-garden-version="version"
/>
`;
